product_attribute_company_favorite/models/product_attribute.py

```diff
@@ -0,0 +1,78 @@
+from odoo import api, fields, models
+
+
+class ProductAttributeIsFavoriteMixin(models.AbstractModel):
+    _name = "product.attribute.favorite.mixin"
+    _description = """Methods used both in product.attribute and
+    product.attribute.value to implement is_favorite functionalities
+    """
+
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        args = list(args or [])
+        args += [("is_favorite", "=", True)]
+        return super()._name_search(
+            name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
+        )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_record_favorite_for_all_companies = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(self.CREATION_ACROSS_COMPANY_CONFIG)
+        )
+        records = super().create(vals_list)
+        if new_record_favorite_for_all_companies:
+            company_ids = (
+                self.env["res.company"].with_context(active_test=False).search([]).ids
+            )
+            for record in records:
+                for company_id in company_ids:
+                    contextual_record = record.with_company(company_id)
+                    contextual_record.is_favorite = True
+        return records
+
+
+class ProductAttribute(models.Model):
+    _inherit = ["product.attribute", "product.attribute.favorite.mixin"]
+    _name = "product.attribute"
+
+    CREATION_ACROSS_COMPANY_CONFIG = (
```

Comment:
I don't see the interest to create a constant for that item. 

why not just use  the following code, and remove the two definition of CREATION_ACROSS_COMPANY_CONFIG ? 

        new_record_favorite_for_all_companies = (
            self.env["ir.config_parameter"]
            .sudo()
            .get_param("product_attribute_company_favorite.new_attribute_favorite_for_all_companies")
        )

Reply:
Because the name of the parameter is not the same for attribute and for attribute values (`new_attribute_favorite_for_all_companies` vs `new_attribute_value_favorite_for_all_companies`)
Or maybe I did not understand your solution ?

Reply:
Is there a use case for these two things to be configured separately?

If there is, a comment would be well-placed here that gives a vague idea what is going on.

Reply:
> Because the name of the parameter is not the same for attribute and for attribute values (new_attribute_favorite_for_all_companies vs new_attribute_value_favorite_for_all_companies)

Ow ! I didn't saw. Well : 
- nitpicking : i'm in favor to remove the both definition and just insert directly the strings in the code. it will be lighter. The constant has to be used only once in the create function. no need to make factorization. (let me know if I'm not clear). 
not a blocking point.

Reply:
The create method is in the mixin, though, and doesn't know whether it's dealing with an attribute or an attribute value :)